### PR TITLE
Feat/layer controls panel

### PR DESCRIPTION
**Pull request by vvandijck**
This pull request includes:
- Show `opacity` changes in MapBox map layers.
- Updated `LayerListControls`. Moved logic into a new `LayerControl` component.
- Prepared a dialog component (`<LayerInfoDialog>`) for when layer info is available.

### CODE REVIEW

File: src/components/LayerListControls/LayerListControls.vue [hunk 1]
```diff
@@ -76,23 +72,49 @@
...
       const { onSortingChange } = useSortable(layers, root, openItems)
 
+      const layerOpacity = (id) => {
+        const activeLayer = Object.assign({}, layers.value.find(item => item.id === id))
+        return activeLayer.opacity
+      }
+
+      const showInfo = (id) => {
+        console.log('show layer info for id:', id)
+        context.root.$store.dispatch('data/setLayerDialogOpen', { open: true }, { root: true })
```
Comment (bob-voorhoede) [inline src/components/LayerListControls/LayerListControls.vue:82]: Is this the only way to access the store in Vue3?

File: src/components/LayerControl/LayerControl.vue [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      },
+      name: {
+        type: String,
+        required: true,
+      },
+      opacity: {
+        type: Number,
+        default: 1,
+        required: true,
+      },
```
Comment (bob-voorhoede) [inline src/components/LayerControl/LayerControl.vue:56]: I think this should probably be reset (and may be other values) when the layer is 'unselected'. But be careful with conflicting store commits that are trying to access the same layer (both layer selection and opacity change trigger something on a `rasterLayer`

File: src/components/LayerListControls/LayerListControls.vue [hunk 3]
```diff
@@ -1,32 +1,25 @@
...
-        </div>
+        <layer-control
+          :id="item.id"
+          :name="item.name"
+          :is-layer="Boolean(item.layer)"
+          :opacity="layerOpacity(item.id)"
+          :parent-ids="item.parentIds.toString()"
+          :selected="selected"
+          @show-info="showInfo"
+          @toggle-layer="toggleLayer"
```
Comment (bob-voorhoede) [inline src/components/LayerListControls/LayerListControls.vue:20]: Is this still doing something?

**Review by bob-voorhoede — CHANGES_REQUESTED** [review]
Very cool to see this working 😎 
Some minor changes and then it's a-okay 🥳

File: src/components/LayerListControls/LayerListControls.vue (hunk 3 above)
Comment (vvandijck) [reply src/components/LayerListControls/LayerListControls.vue:20]: Nope, removed it.

File: src/store/modules/map/index.js [hunk 4]
```diff
@@ -29,6 +29,18 @@ export default {
...
+      const layerToUpdate = state.rasterLayers.find(layer => layer.id === id)
+
+      if (!layerToUpdate) { return }
+
+      layerToUpdate.opacity = opacity
+
+      state.rasterLayers = [
+        ...state.rasterLayers.filter(layer => layer.id !== id),
+        layerToUpdate,
+      ]
```
[the remark this answers is not in the document]
Comment (vvandijck) [reply src/store/modules/map/index.js]: Updated.

File: src/store/modules/map/index.js [hunk 5]
```diff
@@ -31,15 +35,15 @@ export default {
     },
     UPDATE_RASTER_LAYER_OPACITY(state, { id, opacity }) {
       const layerToUpdate = state.rasterLayers.find(layer => layer.id === id)
+      const index = state.rasterLayers.findIndex(layer => layer.id === id)
 
-      if (!layerToUpdate) { return }
+      if (index === -1) {
+        return
+      }
```
Comment (petergoes) [inline src/store/modules/map/index.js:48]: I found the previous method more readable. `layerToUpdate` is still `undefined`, now I have to know what `index` is, and that an index of `-1` means, **non existent**.

The previous version (`if (!layerToUpdate) return`), reads more like english: if no layerToUpdate, return

**Review by petergoes — APPROVED** [review]

**Review by bob-voorhoede — APPROVED** [review]

File: src/store/modules/map/index.js (hunk 5 above)
Comment (vvandijck) [reply src/store/modules/map/index.js]: Reverted.

**Pull request merged**